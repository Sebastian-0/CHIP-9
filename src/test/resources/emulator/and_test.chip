# Test simple ands
ldi A, 85
ldi B, 170
ldi C, 15
ldi D, 240
ldi E, 1
ldi H, 5
ldi L, 24

and A
and B
and C
and D
and E
and H
and L

sout A
sout B
sout C
sout D
sout E
sout H
sout L

ldx HL, 0x1010
ldi (HL), 7
and (HL)

sout (HL)

# Test flags
ldi A, 134
ldi B, 1
ldi C, 128
ldi D, 2

# Zero flag
and B
sout FLAGS

# Negative flag
and C
sout FLAGS

# No flags
and D
sout FLAGS

hcf