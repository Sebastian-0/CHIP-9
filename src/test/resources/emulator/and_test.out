85
0
5
80
1
5
16
5

128
64
0